test/evaluate_integration_test.go

```diff
@@ -0,0 +1,187 @@
+		ok := true
+		for _, key := range keys {
+			_, kok := er.Data[key]
+			ok = ok && kok
+		}
+
+		return ok
+	}
+
+	t.Run("basic evaluate test", test(testCase{
+		testArgs:         a,
+		itemIDs:          itemIDs,
+		query:            createAnnotationQuery(a.prefix),
+		data:             fields.Data().Where(fields.TimeRange(t0, t1)).RollupDuration(time.Hour, time.Monday),
+		timeAggregation:  fields.TimeAggregationAvg,
+		groupAggregation: fields.GroupAggregationAvg,
+		expectedFields:   ef,
+	}))
+
+	for ta := fields.TimeAggregationDefault + 1; ta <= fields.TimeAggregationRate; ta++ {
+		t.Run("time aggregation test type "+fmt.Sprint(ta), test(testCase{
+			testArgs:         a,
+			itemIDs:          itemIDs,
+			query:            createAnnotationQuery(a.prefix),
+			data:             fields.Data().Where(fields.TimeRange(t0, t1)).RollupDuration(time.Hour, time.Monday),
+			timeAggregation:  ta,
+			groupAggregation: fields.GroupAggregationAvg,
+			expectedFields:   ef,
+		}))
+	}
+
+	for ga := fields.GroupAggregationDefault + 1; ga <= fields.GroupAggregationAvg; ga++ {
+		t.Run("group aggregation test type "+fmt.Sprint(ga), test(testCase{
+			testArgs:         a,
+			itemIDs:          itemIDs,
+			query:            createAnnotationQuery(a.prefix),
+			data:             fields.Data().Where(fields.TimeRange(t0, t1)).RollupDuration(time.Hour, time.Monday),
+			timeAggregation:  fields.TimeAggregationAvg,
+			groupAggregation: ga,
+			expectedFields:   ef,
+		}))
+	}
+}
+
+func evaluate(ctx context.Context, client *clarify.Client, itemIDs []string, query fields.ResourceQuery, data fields.DataQuery, timeAggregation fields.TimeAggregation, groupAggregation fields.GroupAggregation) (*clarify.EvaluateResult, error) {
+	f := func(i int, itemID string) fields.EvaluateItem {
+		return fields.EvaluateItem{
+			Alias:           fmt.Sprintf("i%d", i),
+			ID:              itemID,
+			TimeAggregation: timeAggregation,
+			State:           10,
+		}
+	}
+	items := MapIndex(f, itemIDs)
+	groups := []fields.EvaluateGroup{
+		{
+			Alias:            "g1",
+			Query:            query,
+			TimeAggregation:  timeAggregation,
+			GroupAggregation: groupAggregation,
+			State:            10,
+		},
+	}
+	calculations := []fields.Calculation{
+		{Alias: "c1", Formula: "sin(g1)"},
+		{Alias: "c2", Formula: "sin(2*PI*time_seconds/3600)"},
+		{Alias: "c3", Formula: "max(c1,c2)"},
+	}
+	format := views.SelectionFormat{
+		GroupIncludedByType: true,
+	}
+	result, err := client.Clarify().Evaluate(data).
+		Items(items...).
+		Groups(groups...).
+		Calculations(calculations...).
+		Include([]string{"items"}...).
+		Format(format).
+		Do(ctx)
+
+	return result, err
+}
+
+func getListFromResponse(sir *clarify.SelectItemsResult) []string {
+	itemIDs := make([]string, 0, len(sir.Data))
+	for _, d := range sir.Data {
+		itemIDs = append(itemIDs, d.ID)
+	}
+
+	return itemIDs
+}
+
+//lint:ignore U1000 Ignore unused function temporarily for debugging
```

Comment:
Can we remove the lint exceptions?

It's not considered good practice (in Go) to add linter exceptions to code.

Reply:
There are more instances of this, so please do a search.